mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -211,17 +220,79 @@ class KMeans private (
         + " parent RDDs are also uncached.")
     }
 
-    // Compute squared norms and cache them.
-    val norms = data.map(Vectors.norm(_, 2.0))
-    norms.persist()
-    val zippedData = data.zip(norms).map { case (v, norm) =>
-      new VectorWithNorm(v, norm)
+    val zippedData = data.map { x => new VectorWithNorm(x) }
```

Comment:
Below you wrote `map(new VectorWithNorm(_))` which is indeed a little more compact. Not a big deal but if you make another pass you might standardize this.